### ISSUE TITLE
Speed analysis: slower current version 0.2.211026 from previous version 0.2.211012

### ISSUE THREAD

**Issue by ashraful1980** [albertobsd/keyhunt#138]
Hi,
I have run the existing keyhunt software; it was slower from previous version.
The current version result:
└─$ sudo ./keyhunt -m bsgs -f tests/120.txt -t 48 -R -b 120 -q -S -k 8196 -s 30
[+] Version 0.2.211026 Chocolate ¡Beta!, developed by AlbertoBSD
[+] Threads : 48
[+] Random mode
[+] Quiet thread output
[+] K factor 8196
[+] Stats output every 30 seconds
[+] Mode BSGS random
[+] Opening file tests/120.txt
[+] Added 1 points from file
[+] Bit Range 120
[+] -- from : 0x800000000000000000000000000000
[+] -- to   : 0x1000000000000000000000000000000
[+] N = 0xff9ff000000
[+] Bloom filter for 34376515584 elements : 117838.71 MB
[+] Bloom filter for 1718825780 elements : 5891.94 MB
[+] Allocating 26227.00 MB for 1718825780 bP Points
[+] Reading bloom filter from file keyhunt_bsgs_4_34376515584.blm .. Done!
[+] Reading bloom filter from file keyhunt_bsgs_5_1718825780.blm .. Done!
[+] Reading bP Table from file keyhunt_bsgs_2_1718825780.tbl .. Done!
[+] Total 308460792489883729920 keys in 300 seconds: ~1 Ekeys/s (1028202641632945766 keys/s)

Previous version 0.2.211012 result:
![138476302-8ce5df28-c431-4cea-ad27-63c1a387be75](https://user-images.githubusercontent.com/52574658/139533361-472f13da-36d8-4bb5-b1e7-d4ebf9b51daa.png)

I request you please modify current version or publish previous version 0.2.211012.....

**Comment by we-akri** [comment]
Warning: the default n value have a maximun K of 4096 if that value is exceed the program can have an unknow behavior or suboptimal speed. 
fake 1 Ekeys/s they can't even solve puzzle 63

**Comment by kpot87** [comment]
As I know this solved problem with big -k

**Comment by albertobsd** [comment]
I think that the previous speed (11 Exakeys) was wrong, most systems with 128 GB of RAM only run it at some 1 or 2 Exakeys/s and of course this can vary according to the processor speed and number of threads, in the readme there are some warning like this:

**if you exceed the Max value of K the program can have a unknow behavior, the program can have a suboptimal performance, or in the wrong cases you can missing some hits and have an incorrect SPEED.**

Since you are using the default N value the max K for that value is **4096**. To solve this you need increment the N value to `0x400000000000` and half the value of K from 8192 to 4096.

Please use the next combination for your test:

```
-n 0x400000000000 -k 4096
```

### Test if your Speed is REAL or NOT

I just prepare some Set of publickeys to check if your speed is real or not

With  1 Terakey/s the publickey will be found in 2 minutes:
Privatekey: 800000000000068155a43676e00000
Publickey : 022b6a74badcc4c3d8fab7d01ddc1854b9d8f262172789b2aa1bb7fd42cc1b2817

With  5 Terakeys/s the publickey will be found in 2 minutes
Privatekey: 8000000000002086ac351052600000
Publickey : 024cf9e44f808e7b0bbb12a57ff63e3a8407cba1816f5e31e815d33d70e4a95a7f

With 10 Terakeys/s the publickey will be found in 2 minutes
Privatekey: 800000000000410d586a20a4c00000
Publickey : 02ee0cf78d13b4aae9c8777a0f93dff7f5be3855bd2c0f85370f861c69bb5b533a

Select one publickey that fit to your current speed save it in a file `testpublickey.txt` and test it with:

```
./keyhunt -m bsgs -f testpublickey.txt -b 120 -n 0x400000000000 -k 4096
```

The publickeys should be found in some 2 minutes after the load of the files

Change your n or k values according to your current memory and remember not exceed the k value of each N please check the table https://github.com/albertobsd/keyhunt#valid-n-and-k-values

If you need to exceed it please increment the N value and half the K value this will be equivalent.

Please remember that this software is still a Beta version and the development of it is free or charge.

**Comment by ashraful1980** [comment]
Hi,
Already i have checked version 0.2.211012 as following; it is found correct result no bug yet.

![image](https://user-images.githubusercontent.com/52574658/140599427-9b74f591-e830-4609-a7a3-6945377ab341.png)

**Comment by ashraful1980** [comment]
Evaluation of the keyhunt speed:
![Keyhunt_Speed](https://user-images.githubusercontent.com/52574658/140729804-8d86b9a8-c60a-4730-ba90-aca7b813e9f6.jpg)

**Comment by albertobsd** [comment]
Check one of those cases please https://github.com/albertobsd/keyhunt#is-my-speed-real

if you found the selected key in that time then the speed is OK, if no, the speed is WRONG
I bet that the speed is WRONG

**Comment by albertobsd** [comment]
I only will check and solve this case if the program find the next cases https://github.com/albertobsd/keyhunt#is-my-speed-real  and the next

8000000000004FFFFFFFFFFFF00000
8000000000007FFFFFFFFFFFFFFFFF
800000000000800000000000000000
800000000000800000000000000001

Also only if those are found in a decent time.

Regards!

**Comment by albertobsd** [comment]
Also please check this BUG https://github.com/albertobsd/keyhunt/issues/130 that bug was solved but it STILL EXIST in the version 0.2.211012.
